connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceAdd.java

```diff
@@ -76,82 +77,91 @@
     @Override
     protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource) throws  OperationFailedException {
         super.populateModel(context, operation, resource);
-        final ModelNode model = resource.getModel();
-        boolean enabled = ! operation.hasDefined(ENABLED.getName()) || ENABLED.resolveModelAttribute(context, model).asBoolean();
+        DataSourceStatisticsService.registerStatisticsResources(resource);
+        boolean enabled = ! operation.hasDefined(ENABLED.getName()) || ENABLED.resolveModelAttribute(context, resource.getModel()).asBoolean();
         if (enabled) {
             context.addStep(new DataSourceEnable(this instanceof XaDataSourceAdd), OperationContext.Stage.MODEL);
```

Comment:
If the DS is added after boot, DataSourceEnable L82 is going to change the attribute value from an expression to 'true' and the user's setting will be lost.